components/ai_chat/core/browser/engine/conversation_api_v2_client.cc

```diff
@@ -40,6 +68,67 @@ net::NetworkTrafficAnnotationTag GetNetworkTrafficAnnotationTag() {
 
 }  // namespace
 
+// static
+base::Value::List ConversationAPIV2Client::SerializeOAIMessages(
+    std::vector<OAIMessage> messages) {
+  static constexpr auto kTypeMap =
+      base::MakeFixedFlatMap<ExtendedContentBlockType, std::string_view>(
+          {{ExtendedContentBlockType::kText, "text"},
+           {ExtendedContentBlockType::kImage, "image_url"},
+           {ExtendedContentBlockType::kPageExcerpt, "brave-page-excerpt"},
+           {ExtendedContentBlockType::kChangeTone, "brave-request-change-tone"},
+           {ExtendedContentBlockType::kParaphrase, "brave-request-paraphrase"},
+           {ExtendedContentBlockType::kImprove,
+            "brave-request-improve-excerpt-language"},
+           {ExtendedContentBlockType::kShorten, "brave-request-shorten"},
+           {ExtendedContentBlockType::kExpand, "brave-request-expansion"}});
+
+  base::Value::List serialized_messages;
+  for (auto& message : messages) {
+    base::Value::Dict message_dict;
+
+    // Set role
+    message_dict.Set("role", std::move(message.role));
+
+    // Content blocks
+    base::Value::List content_list;
+    for (auto& extended_content_block : message.content) {
+      base::Value::Dict content_block_dict;
+
+      // Set type
+      auto type_it = kTypeMap.find(extended_content_block.type);
+      CHECK(type_it != kTypeMap.end());
```

Comment:
similar to https://github.com/brave/brave-core/pull/32663/files#r2591062481, can we just ignore types we don't support instead of crashing it or return a empty list or `nullopt` with `std::optional`?

Reply:
https://github.com/brave/brave-core/pull/32663/commits/fc4ddbaef89521628a54255888bb702ca9956baf